src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -34,4 +39,116 @@ public function __construct(ContainerBuilder $container, FileLocatorInterface $l
 
         parent::__construct($locator);
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setCurrentDir($dir)
+    {
+        $this->currentDir = $dir;
+        parent::setCurrentDir($dir);
+    }
+
+    public function registerClasses(Definition $prototype, $psr4Glob, $directory)
```

Comment:
I seriously hope that `Parameter Type Widening` is not performed when strict-types are used (in the child class file) 😑 this would be a big mistake otherwise.